bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/channel/ChannelTypeResource.java

```diff
@@ -0,0 +1,144 @@
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionDTO;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionDTOMapper;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionParameterDTO;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionParameterGroupDTO;
+import org.eclipse.smarthome.core.thing.dto.ChannelTypeDTO;
+import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeRegistry;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.io.rest.LocaleUtil;
+import org.eclipse.smarthome.io.rest.RESTResource;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * Provides access to ChannelType via REST.
+ *
+ * @author Chris Jackson - Initial contribution
+ */
+@Path(ChannelTypeResource.PATH_CHANNEL_TYPES)
+@Api
+public class ChannelTypeResource implements RESTResource {
+
+    /** The URI path to this resource */
+    public static final String PATH_CHANNEL_TYPES = "channel-types";
+
+    private ChannelTypeRegistry channelTypeRegistry;
+    private ConfigDescriptionRegistry configDescriptionRegistry;
+
+    protected void setChannelTypeRegistry(ChannelTypeRegistry channelTypeRegistry) {
+        this.channelTypeRegistry = channelTypeRegistry;
+    }
+
+    protected void unsetChannelTypeRegistry(ChannelTypeRegistry channelTypeRegistry) {
+        this.channelTypeRegistry = null;
+    }
+
+    protected void setConfigDescriptionRegistry(ConfigDescriptionRegistry configDescriptionRegistry) {
+        this.configDescriptionRegistry = configDescriptionRegistry;
+    }
+
+    protected void unsetConfigDescriptionRegistry(ConfigDescriptionRegistry configDescriptionRegistry) {
+        this.configDescriptionRegistry = null;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gets all available channel types.", response = ChannelTypeDTO.class, responseContainer = "Set")
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response getAll(@HeaderParam("Accept-Language") @ApiParam(value = "Accept-Language") String language) {
+        Locale locale = LocaleUtil.getLocale(language);
+        Set<ChannelTypeDTO> channelTypeDTOs = convertToChannelTypeDTOs(channelTypeRegistry.getChannelTypes(locale),
+                locale);
+        return Response.ok(channelTypeDTOs).build();
+    }
+
+    @GET
+    @Path("/{channelTypeUID}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gets channel type by UID.", response = ChannelTypeDTO.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Channel type with provided channelTypeUID does not exist."),
+            @ApiResponse(code = 204, message = "No content") })
```

Comment:
I think this should be a 404 - I see that you have copied it from the ThingTypeResource and it does the same. But that resource was apparently forgotten in the recent refactoring that changed exactly this: https://github.com/eclipse/smarthome/pull/505/files#diff-ff96ef1143148ac59cb004fabda9cb73L142
So maybe you can fix the ThingTypeResource along the way as well :-)